front/app/modules/free/native_surveys/admin/containers/projects/surveys/edit/index.tsx

```diff
@@ -23,29 +36,86 @@ const StyledRightColumn = styled(RightColumn)`
   overflow-y: auto;
 `;
 
-export const SurveyEdit = () => (
-  <Box
-    display="flex"
-    flexDirection="column"
-    w="100%"
-    zIndex="10000"
-    position="fixed"
-    bgColor={colors.adminBackground}
-    h="100vh"
-    data-testid="surveyBuilderPage"
-  >
-    <FocusOn>
-      <SurveyBuilderTopBar />
-      <Box mt={`${stylingConsts.menuHeight}px`} display="flex">
-        <SurveyBuilderToolbox />
-        <StyledRightColumn>
-          <Box width="1000px" bgColor="#ffffff" minHeight="300px" />
-        </StyledRightColumn>
-        <SurveyBuilderSettings />
-      </Box>
-    </FocusOn>
-  </Box>
-);
+export const SurveyEdit = () => {
+  const [selectedField, setSelectedField] = useState<
+    IFlatCustomField | undefined
+  >(undefined);
+  const { projectId } = useParams() as { projectId: string };
+
+  const { surveyCustomFields, setSurveyCustomFields } = useSurveyCustomFields({
+    projectId,
+  });
+
+  const closeSettings = () => {
+    setSelectedField(undefined);
+  };
+
+  const handleDelete = async (fieldId: string) => {
+    if (!isNilOrError(surveyCustomFields)) {
+      const newFields = surveyCustomFields.filter(
+        (field) => field.id !== fieldId
+      );
+      setSurveyCustomFields(newFields);
+      closeSettings();
+    }
+  };
+
+  const onAddField = (field: IFlatCreateCustomField) => {
+    if (!isNilOrError(surveyCustomFields)) {
+      setSurveyCustomFields(
+        surveyCustomFields.concat([field as IFlatCustomField])
```

Comment:
Just curious here again if we need to use the "as" keyword

Reply:
It should be fine in this case but I have removed the one above and added a TODO here for myself to improve it.